### DISCUSSION TITLE
[integration_test] Test firebase auth (email link or google sign in)

### DISCUSSION THREAD

**Discussion by tzvc** [firebase/flutterfire#6293] [Q&A]
Hi there,

I'm adding integration testing (using the `integration_test` package) to my app but I am running into a problem.

Let me explain. The first step when my app launch is authentication for which I have 3 options: firebase email link, firebase google sign in, and firebase facebook sign in.

What is blocking me is that all these sign in methods require actions outside of the main app dart code and thus are not accessible by flutter driver.

Am I missing something here? And if not how should that case be handled?

**Comment by markusaksli-nc** [comment]
If you are using firebase emulator during testing you should be able to solve this by scripting the whole process (start emulator, start integration test, wait for emulator to give you the link, request it etc).

This sounds pretty fragile and could cause flaky results though so maybe someone knows a better way.

For regular tests you could look at https://github.com/FirebaseExtended/flutterfire/blob/master/packages/firebase_auth/firebase_auth/test/firebase_auth_test.dart for some ideas
